.github/workflows/wheel_tests_and_release.yml

```diff
@@ -0,0 +1,213 @@
+name: Build Wheels and Release
+
+on:
+  push:
+    branches:
+      - cibuildwheel
+      - master
+
+    tags:
+      - 'v*'
+      - 'buildwheels*'
+
+  pull_request:
+
+jobs:
+  build_linux_wheels:
+    name: Build ${{ matrix.cibw_python }}-${{ matrix.cibw_libc }}_${{ matrix.cibw_arch }} wheel on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+
+        cibw_python: ["cp37", "cp38", "cp39", "cp310"]
+        # SciPy and NumPy don't support musllinux
+        cibw_libc: ["manylinux"]
+        cibw_arch: ["x86_64", "i686", "aarch64"]
+
+        exclude:
+          # SciPy and NumPy don't support 32-bit Linux from Python 3.10 and onwards
+          - cibw_python: "cp310"
+            cibw_arch: "i686"
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.9'
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: arm64
+
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel
+
+      - name: Build the wheel
+        run: |
+          python -m cibuildwheel --output-dir dist
+        env:
+          CIBW_BUILD: ${{ matrix.cibw_python }}-${{ matrix.cibw_libc }}_${{ matrix.cibw_arch }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: ./dist/*.whl
```

Comment:
can we have some type of if clause to disable this on pull_requests? Otherwise each time a user makes a commit in a PR we will generate and store wheel artifacts, which seems wasteful and not very useful (don't recall offhand how long GitHub retains the artifacts).

Reply:
Done, I think. Bit hard to test for now. I believe retention is 90 days?